src/generateTasks.js

```diff
@@ -1,26 +1,23 @@
 'use strict'
 
+const path = require('path')
 const minimatch = require('minimatch')
-
-const readConfigOption = require('./readConfigOption')
+const getConfig = require('./getConfig')
+const resolveGitDir = require('./resolveGitDir')
 
 module.exports = function generateTasks(config, files) {
-  const linters = config.linters !== undefined ? config.linters : config
-  const resolve = file => files[file]
+  const normalizedConfig = getConfig(config) // Ensure we have a normalized config
```

Comment:
Can we avoid repeatedly executing `getConfig`? I suggest we set a property `_normalized` to `true` which can be used in the following ways:
- Use the property for an early return in `getConfig`. 
- Check that the config has been normalized in all places other than the first one(`runAll`) and throw an error if not.

Reply:
Good suggestions. You want to work on this?

Reply:
😄 Okay. I usually have to hunt around for finding open source work. Believe it or not, I am very grateful to you for sending some work my way.